packages/web/src/js/Dropdown.ts

```diff
@@ -0,0 +1,203 @@
+import { computePosition, flip, shift, offset, autoUpdate } from '@floating-ui/dom';
+import BaseComponent from './BaseComponent';
+import { enableToggleTrigger } from './utils/ComponentFunctions';
+import EventHandler from './dom/EventHandler';
+import SelectorEngine from './dom/SelectorEngine';
+
+enum placementKeys {
+  'top-start' = 'top-start',
+  'top-end' = 'top-end',
+  'right-start' = 'right-start',
+  'right-end' = 'right-end',
+  'bottom-start' = 'bottom-start',
+  'bottom-end' = 'bottom-end',
+  'left-start' = 'left-start',
+  'left-end' = 'left-end',
+}
+
+interface DropdownStateProps {
+  open: boolean;
+}
+
+interface DropdownOptionsProps {
+  placement: keyof typeof placementKeys;
+  offset: number;
+  padding: number;
+  autoClose: boolean;
+}
+
+type autoUpdateOptionsType = {
+  ancestorScroll?: boolean;
+  ancestorResize?: boolean;
+  elementResize?: boolean;
+  animationFrame?: boolean;
+};
+
+const NAME = 'dropdown';
+const DATA_KEY = `${NAME}`;
+const EVENT_KEY = `.${DATA_KEY}`;
+const EVENT_HIDE = `hide${EVENT_KEY}`;
+const EVENT_HIDDEN = `hidden${EVENT_KEY}`;
+const EVENT_SHOW = `show${EVENT_KEY}`;
+const EVENT_SHOWN = `shown${EVENT_KEY}`;
+const ARIA_EXPANDED_ATTRIBUTE = 'aria-expanded';
+const ARIA_CONTROLS_ATTRIBUTE = 'aria-controls';
+const CLASSNAME_DROPDOWN = 'dropdown';
+export const CLASSNAME_EXPANDED = 'is-expanded';
+export const CLASSNAME_OPEN = 'is-open';
+
+const clickOutsideElement = (target: Element, event: Event) => !event.composedPath().includes(target);
+
+class Dropdown extends BaseComponent {
+  target: HTMLElement | null | undefined;
+  anchor: HTMLElement | null | undefined;
+  state: DropdownStateProps;
+  options: DropdownOptionsProps;
+  autoUpdateOptions: autoUpdateOptionsType;
+
+  constructor(element: SpiritElement) {
+    super(element);
+    this.target = this.findTargetElement();
+    this.anchor = this.findAnchorElement();
+    this.state = {
+      open: false,
+    };
+    this.options = {
+      placement: placementKeys['bottom-start'],
+      offset: 0,
+      padding: 0,
+      autoClose: true,
+    };
+    this.autoUpdateOptions = {
+      ancestorScroll: true,
+      ancestorResize: true,
+      elementResize: true,
+      animationFrame: false,
+    };
+  }
+
+  static get NAME() {
+    return NAME;
+  }
+
+  getOptions() {
+    const options = { ...this.options };
+    const dataset = this.element?.dataset;
+    const optionsPlacement = dataset?.placement;
+    const optionsOffset = dataset?.offset;
+    const optionsPadding = dataset?.padding;
+    const optionsAutoClose = dataset?.autoclose;
+    if (optionsPlacement) options.placement = optionsPlacement;
+    if (optionsOffset) options.offset = Number(optionsOffset);
+    if (optionsPadding) options.padding = Number(optionsPadding);
+    if (optionsAutoClose) options.autoClose = Boolean(!optionsAutoClose);
+
+    return options;
+  }
+
+  findAnchorElement() {
```

Comment:
Please choose a better name. `AnchorElement` indicates that you are working with the `a` tag. But you are not.

Reply:
Ok, i changed this to `this.reference` and `this.findReferenceElement()`